libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/metrics/MapboxMetricsReporter.kt

```diff
@@ -0,0 +1,101 @@
+package com.mapbox.services.android.navigation.v5.internal.navigation.metrics
+
+import android.content.Context
+import com.google.gson.Gson
+import com.mapbox.android.telemetry.AppUserTurnstile
+import com.mapbox.android.telemetry.MapboxTelemetry
+import com.mapbox.navigation.metrics.DirectionsMetrics
+import com.mapbox.navigation.metrics.MetricsObserver
+import com.mapbox.navigation.metrics.NavigationMetrics
+import com.mapbox.services.android.navigation.v5.internal.navigation.InitialGpsEvent
+import com.mapbox.services.android.navigation.v5.internal.navigation.RouteRetrievalEvent
+import java.util.concurrent.atomic.AtomicReference
+
+internal object MapboxMetricsReporter : DirectionsMetrics, NavigationMetrics {
+
+    private lateinit var gson: Gson
+    private lateinit var mapboxTelemetry: MapboxTelemetry
+    private val metricsObserver: AtomicReference<MetricsObserver?> = AtomicReference()
+
+    fun init(
+        context: Context,
+        accessToken: String,
+        userAgent: String,
+        gson: Gson
+    ) {
+        this.mapboxTelemetry = MapboxTelemetry(context, accessToken, userAgent)
+        this.mapboxTelemetry.enable()
+        this.gson = gson
+    }
+
+    fun disable() {
+        mapboxTelemetry.disable()
+    }
+
+    override fun toggleLogging(isDebugLoggingEnabled: Boolean) {
+        mapboxTelemetry.updateDebugLoggingEnabled(isDebugLoggingEnabled)
+    }
+
+    override fun arriveEvent(eventName: String, eventJsonString: String) {
+        val event = fromGson(eventJsonString, NavigationArriveEvent::class.java)
+        // TODO: Move metricsObserver call to separate thread
+        metricsObserver.get()?.onJsonStringMetricUpdated(eventName, eventJsonString)
```

Comment:
let's debate if we should stick with observer here or just simply roll with default impl in order to keep things simple.